Lombiq.HelpfulLibraries/Libraries/Mvc/TypedRoute.cs

```diff
@@ -0,0 +1,135 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using OrchardCore.Admin;
+using OrchardCore.Environment.Extensions;
+using OrchardCore.Modules.Manifest;
+using OrchardCore.Mvc.Core.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Net;
+using System.Reflection;
+
+namespace Lombiq.HelpfulLibraries.Libraries.Mvc
+{
+    public class TypedRoute
+    {
+        private readonly List<KeyValuePair<string, string>> _arguments;
+
+        private readonly Lazy<bool> _isAdminLazy;
+        private readonly Lazy<string> _routeLazy;
+
+        public TypedRoute(
+            Type controller,
+            MethodInfo action,
+            IEnumerable<KeyValuePair<string, string>> arguments,
+            ITypeFeatureProvider typeFeatureProvider = null)
+        {
+            string area;
+            _arguments = arguments is List<KeyValuePair<string, string>> list ? list : arguments.ToList();
+            var areaPair = _arguments.FirstOrDefault(pair => pair.Key.EqualsOrdinalIgnoreCase("area"));
+            if (areaPair.Value is { } areaArgumentValue)

```

Comment:
It's an empty [property pattern](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/patterns#property-pattern). It's the same as `!= null`. It's recommended to use this format in c# 8 and above for the same reason we use `var` instead of the type name as well.